modules/exceptions.py

```diff
@@ -0,0 +1,6 @@
+class InvalidResponseContent(Exception):
```

Comment:
I just realised it would be useful to use a single base class for all API exceptions. E.g. `ApiError`. That way we can check for any deliberate `ApiError`, as opposed to random other errors - as might often be useful.